### PR TITLE
Create _P222_DebouncedCounter.ino

### PR DESCRIPTION
**Pull request by oerweber**
Started development of a improved generic impulse counter module. The core module  _P003_Pulse has a problem counting RISING and FALLING edges when the signal bounces during transition. This is caused by the kind of implementation of the debounce function (delay type) which is not able to decide beween bounced edges and real edges.

This approach will try to implement a timer based solution as it is used in 8-bit controllers very successful.

### CODE REVIEW

File: _P222_DebouncedCounter.ino [hunk 1]
```diff
@@ -22,13 +22,15 @@ void Plugin_222_pulse_interrupt4() ICACHE_RAM_ATTR;
...
 byte Plugin_222_pulsePin[TASKS_MAX];
 byte Plugin_222_pulseMode[TASKS_MAX];
 byte Plugin_222_ct0[TASKS_MAX];
 byte Plugin_222_ct1[TASKS_MAX];
-byte Plugin_222_key_state[TASKS_MAX];
+byte Plugin_222_prescaler[TASKS_MAX];
+byte Plugin_222_prescaler_cnt[TASKS_MAX];
+byte Plugin_222_signal_state[TASKS_MAX];
 byte Plugin_222_signal_falling[TASKS_MAX];
 byte Plugin_222_signal_rising[TASKS_MAX];
```
Comment (TD-er) [inline _P222_DebouncedCounter.ino]: This does allocate `TASK_MAX` * (9 bytes + 4 longs = 25 bytes) = 300 bytes.
That's a lot of memory allocated on sys stack regardless whether this is being used.

File: _P222_DebouncedCounter.ino [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+#define PLUGIN_ID_222         222
+#define PLUGIN_NAME_222       "Generic - Debounced Counter"
+#define PLUGIN_VALUENAME1_222 "Count"
+#define PLUGIN_VALUENAME2_222 "Total"
+#define PLUGIN_VALUENAME3_222 "Time"
+
+void Plugin_222_pulse_interrupt1() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt2() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt3() ICACHE_RAM_ATTR;
+void Plugin_222_pulse_interrupt4() ICACHE_RAM_ATTR;
```
Comment (TD-er) [inline _P222_DebouncedCounter.ino]: I don't see these functions in your code.
Also doesn't it make more sense to have 1 function and just look at the internal registers to fetch all pin states to simply see what's changed and register those as flags that need to be taken care of in the normal loop?
To fetch all pin states, it takes a single read. Compare it with the previous known state is a single XOR operation and then store the difference in a global variable.
If not 0, it means something was changed which must then be handled with in the 50/sec loop.
You may also apply a set mask to only trigger on the pins you're interested in.

**Comment by glsf91** [comment]
Why is this plugin removed from the playground?
It was there a time ago.

**Comment by TD-er** [comment]
> Why is this plugin removed from the playground?
> It was there a time ago.

More like I forgot to merge it....

**Pull request merged**